api/v1/lib/extras/scheduler/controller/controller.go

```diff
@@ -78,7 +76,9 @@ func (_ *controllerImpl) Run(config Config) (lastErr error) {
 	for !config.Context.Done() {
 		frameworkID := config.Context.FrameworkID()
 		if config.Framework.GetFailoverTimeout() > 0 && frameworkID != "" {
-			subscribe.Subscribe.FrameworkInfo.ID = &mesos.FrameworkID{Value: frameworkID}
+			frameworkProto := &mesos.FrameworkID{Value: frameworkID}
+			subscribe.Subscribe.FrameworkInfo.ID = frameworkProto
```

Comment:
would be nice to consolidate these lines and say something like `subscribe.WithFrameworkID(frameworkID)`